baxter_collaboration_lib/include/robot_utils/utils.h

```diff
@@ -107,6 +107,16 @@ void setPosition(geometry_msgs::Pose& pose, float x, float y, float z);
  */
 void setOrientation(geometry_msgs::Pose& pose, float x, float y, float z, float w);
 
+/**
+ * replaces all instances of a substring with specified substring
+ * @param target      string containing the substring
+ * @param replaceThis substring to be replaced
+ * @param replaceWith substring to replace with
+ *
+ * return N/A
+ */
+void subStringReplace(std::string& target, std::string replaceThis, std::string replaceWith);
```

Comment:
What if you change `toString(vector)` into a for loop that cycles through the elements and does individual `toString(double)`? Overall, this is a very minor function, whose behavior is not critical since it is only used for printouts. I would try to move on with this as fast as possible and start testing more important things.